### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by NNieuwenhuis**

### CODE REVIEW

File: src/app/module-overview-component/module-overview.component.ts [hunk 1]
```diff
@@ -9,15 +10,34 @@ import { BackendService} from '../backend.service';
...
+  curricula: Curriculum[];
+  selectedCurriculum: number;
 
-  constructor( private bms: BackendService) {}
+  constructor(private backendService: BackendService) {
+  }
+
+  onSelect(curriculum: Curriculum): void {
+    this.selectedCurriculum = curriculum.id;
+    console.log(this.selectedCurriculum);
```
Comment (Lehren) [inline src/app/module-overview-component/module-overview.component.ts]: Do you need these console logs

File: src/app/module-overview-component/module-overview.component.ts [hunk 2]
```diff
@@ -9,15 +10,34 @@ import { BackendService} from '../backend.service';
...
+
+  getCurriculum(): void {
+    this.backendService.getCurricula().subscribe(curricula => this.curricula = curricula);
   }
 
   ngOnInit(): void {
-    this.getSemesters();
+    this.getCurriculum();
+    // this.selectedCurriculum = 'BI';
+    // this.getSemesters();
```
Comment (Lehren) [inline src/app/module-overview-component/module-overview.component.ts]: Remove commented code

**Review by Lehren — CHANGES_REQUESTED** [review]

File: src/app/backend.service.ts [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
 import {Observable} from 'rxjs/Observable';
 import {HttpClient} from '@angular/common/http';
 
 import 'rxjs/add/operator/map';
+import {CurriculaResponse} from './backend-responses/CurriculaResponse';
 
 @Injectable()
 export class BackendService {
   private headers = new Headers({'Content-type': 'application/json'});
   private semestersUrl = 'http://172.17.0.1:8080/fmms/curriculum/BI/semesters';
```
Comment (lukeelten) [inline src/app/backend.service.ts]: What is this?

File: src/app/backend.service.ts [hunk 4]
```diff
@@ -1,24 +1,34 @@
...
-      .map(data => data.semesters);
+  getCurricula(): Observable<Curriculum[]> {
+    return this.http.get<CurriculaResponse>(this.curriculaUrl)
+      .map(data => data.curricula);
   }
 
+// TO CHECK might need to be casted to string before appending it.
+// I'm not that familair with angular. If it works you can remove this comment
+  getSemesters(id: number): Observable<Semester[]> {
+    const semestersUrl = 'http://localhost:8080/fmms/curriculum/' + id + '/semesters';
```
Comment (lukeelten) [inline src/app/backend.service.ts]: Localhost will not work inside Docker

File: src/app/backend.service.ts [hunk 5]
```diff
@@ -12,8 +12,7 @@ import {CurriculaResponse} from './backend-responses/CurriculaResponse';
 @Injectable()
 export class BackendService {
   private headers = new Headers({'Content-type': 'application/json'});
-
-  private curriculaUrl = 'http://localhost:8080/fmms/curricula';
+  private semestersUrl = 'http://172.17.0.1:8080/fmms/curriculum/BI/semesters';
```
Comment (Lehren) [inline src/app/backend.service.ts]: What is this.
We don't use BI or SE anymore in the URL.

File: src/app/backend.service.ts [hunk 6]
```diff
@@ -1,24 +1,34 @@
...
 
+// TO CHECK might need to be casted to string before appending it.
+// I'm not that familair with angular. If it works you can remove this comment
+  getSemesters(id: number): Observable<Semester[]> {
+    const semestersUrl = 'http://localhost:8080/fmms/curriculum/' + id + '/semesters';
+    return this.http.get<CurriculumResponse>(semestersUrl)
+      .map(data => data.semesters);
+  }
 
   private handleError(error: any): Promise<any> {
```
Comment (lukeelten) [inline src/app/backend.service.ts:33]: Where is this used for?

File: src/app/backend.service.ts [hunk 7]
```diff
@@ -1,24 +1,34 @@
...
 
-  constructor(private http: HttpClient) { }
+  constructor(private http: HttpClient) {
+  }
 
-  getSemesters(): Observable<Semester[]> {
-    return this.http.get<CurriculumResponse>(this.semestersUrl)
-      .map(data => data.semesters);
+  getCurricula(): Observable<Curriculum[]> {
+    return this.http.get<CurriculaResponse>(this.curriculaUrl)
```
Comment (lukeelten) [inline src/app/backend.service.ts:21]: Where can I find this.curriculaUrl?

**Review by lukeelten — CHANGES_REQUESTED** [review]

**Review by Lehren — APPROVED** [review]

**Review by lukeelten — APPROVED** [review]

**Pull request merged**